consensus/src/chained_bft/liveness/proposal_generator.rs

```diff
@@ -108,6 +121,12 @@ impl<T: Payload> ProposalGenerator<T> {
             "The epoch has already ended,a proposal is not allowed to generated"
         );
 
+        if self.block_store.root().id() != hqc.certified_block().id()
```

Comment:
what's the relevance here of this check? would we ever get into a position where this is true but the block has reconfigurations? how would we handle that case?

Reply:
this was used to filter out genesis or committed reconfiguration, let me see if I can get rid of it